ml-package-versions.yml

```diff
@@ -0,0 +1,162 @@
+sklearn:
+  package_info:
+    pip_release: "scikit-learn"
+    pip_dev: |
+      pip install git+https://github.com/scikit-learn/scikit-learn.git@master
+
+  models:
+    minimum: "0.20.4"
+    maximum: "0.23.2"
+    run: |
+      pytest tests/sklearn/test_sklearn_model_export.py --large
+
+  autologging:
+    minimum: "0.20.4"
+    maximum: "0.23.2"
+    requirements: ["matplotlib"]
+    run: |
+      pytest tests/sklearn/test_sklearn_training_session.py --large
+      pytest tests/sklearn/test_sklearn_autolog.py --large
+
+pytorch:
+  package_info:
+    pip_release: "torch"
+    pip_dev: |
+      pip install --pre torch -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html
+
+  models:
+    minimum: "1.4.0"
+    maximum: "1.7.0"
+    requirements:
+      # https://github.com/pytorch/vision#installation
+      "== 1.4.0": ["torchvision==0.5.0", "scikit-learn"]
+      "== 1.5.0": ["torchvision==0.6.0", "scikit-learn"]
+      "== 1.5.1": ["torchvision==0.6.1", "scikit-learn"]
+      "== 1.6.0": ["torchvision==0.7.0", "scikit-learn"]
+      "> 1.6.0": ["torchvision", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+  autologging:
+    minimum: "1.4.0"
+    maximum: "1.7.0"
+    requirements:
+      "== 1.4.0": ["torchvision==0.5.0", "pytorch-lightning", "scikit-learn"]
+      "== 1.5.0": ["torchvision==0.6.0", "pytorch-lightning", "scikit-learn"]
+      "== 1.5.1": ["torchvision==0.6.1", "pytorch-lightning", "scikit-learn"]
+      "== 1.6.0": ["torchvision==0.7.0", "pytorch-lightning", "scikit-learn"]
+      "> 1.6.0": ["torchvision", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+tensorflow:
+  package_info:
+    pip_release: "tensorflow"
+    pip_dev: |
+      pip install tf-nightly
+
+  models:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    run: |
+      pytest tests/tensorflow/test_tensorflow2_model_export.py --large
+
+  autologging:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    requirements:
+      "< 2.2": ["h5py<3.0"]
+    run: |
+      pytest --verbose tests/tensorflow_autolog/test_tensorflow2_autolog.py --large
+
+  tf1:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    # Avoid fetching versions up until the latest
+    until_latest: False
+    # Avoid running tests on `tf-nightly`
+    include_dev: False
```

Comment:
Instead of nesting `tf1` under `tensorflow` and introducing additional YAML fields (`include_latest` and `include_dev`), can we instead use two top-level YAML entries: one for `tensorflow1.x` and the other for `tensorflow2.x`? 

To solve the `until_latest` challenge, I think it makes sense to adopt the following convention:

```
For maximum version x.a.b, we'll test up to x.y.z, where x.y.z is the latest minor version that shares the same major version as the configured maximum.
```

This way, for lightgbm, which spans multiple major versions, we'll test from 2.x to 3.x. For TensorFlow 1, which uses a single major version (`1`), we'll only test 1.x.

Regarding `include_dev`, I think we can drop this and only run dev tests if `package_info` specifies `pip_dev`. That way, the `tensorflow1.x` tests can just omit `pip_dev`.

Reply:
Nice approach. I made the change!